src/main/java/org/opensearch/neuralsearch/processor/combination/ArithmeticMeanScoreCombinationTechnique.java

```diff
@@ -26,8 +44,13 @@ public class ArithmeticMeanScoreCombinationTechnique implements ScoreCombination
     public float combine(final float[] scores) {
         float combinedScore = 0.0f;
         int count = 0;
-        for (float score : scores) {
+        for (int i = 0; i < scores.length; i++) {
+            float score = scores[i];
             if (score >= 0.0) {
+                // apply weight for this sub-query if it's set for particular sub-query
+                if (i < weights.size()) {
+                    score = (float) (score * weights.get(i));
+                }
                 combinedScore += score;
                 count++;
```

Comment:
Maybe `count += weight`? I think weighted arithmetic mean wants to divide by total weight of the scores, not count. Otherwise you throw off your scaling, e.g. I have weights [300, 650, 50]. Doesn't matter all that much but I guess it does raise the question of what happens if the provided weights are way out of scale with the default weight (1.0)

Reply:
@HenryL27 it needs to divide by the count. "mean" always mean divide by the number of entities you have. 

weights: here is a representation to say how much importance we want to give for a score of the document coming from a specific query. which in turn means how important the results of the query belong in the final result set.

Please correct me if there is another definition of arithmetic mean

Reply:
@martin-gaievski https://en.wikipedia.org/wiki/Weighted_arithmetic_mean

Reply:
nice, let me correct implementation, it should be divided by sum of weights as @HenryL27 mentioned

Reply:
Thank you for pointing this out, we implemented "normalized weighted linear combination", I will change implementation accordingly